lib/trino-orc/src/main/java/io/trino/orc/OrcOutputBuffer.java

```diff
@@ -85,23 +85,24 @@ public OrcOutputBuffer(CompressionKind compression, int maxBufferSize)
 
         compressedOutputStream = new ChunkedSliceOutput(MINIMUM_OUTPUT_BUFFER_CHUNK_SIZE, MAXIMUM_OUTPUT_BUFFER_CHUNK_SIZE);
 
-        if (compression == CompressionKind.NONE) {
-            this.compressor = null;
-        }
-        else if (compression == CompressionKind.SNAPPY) {
-            this.compressor = new SnappyCompressor();
-        }
-        else if (compression == CompressionKind.ZLIB) {
-            this.compressor = new DeflateCompressor();
-        }
-        else if (compression == CompressionKind.LZ4) {
-            this.compressor = new Lz4Compressor();
-        }
-        else if (compression == CompressionKind.ZSTD) {
-            this.compressor = new ZstdCompressor();
-        }
-        else {
-            throw new IllegalArgumentException("Unsupported compression " + compression);
+        switch (compression) {
+            case NONE:
+                this.compressor = null;
+                break;
```

Comment:
I would extract this as method so you can replace `break` with `return`

Reply:
IMO this shouldn't be part of this PR